File: level 1/기출문제/실패율/solution.py
def solution(N, stages):
    user = len(stages)
    answer = []
    
    for i in range(1, N+1):
        if user != 0:
            cnt = stages.count(i)
            answer.append([i, cnt / user])
            user -= cnt
        else:
            answer.append([i, 0])
    new = sorted(answer, key=lambda x: x[1], reverse=True)
    
    return [a[0] for a in new]


def solution(N, stages):
    result = {}
    denominator = len(stages)
    for stage in range(1, N+1):
        if denominator != 0:
            count = stages.count(stage)
            result[stage] = count / denominator
            denominator -= count
        else:
            result[stage] = 0
    return sorted(result, key=lambda x : result[x], reverse=True)

File: level 1/기출문제/실패율/solution.js
function solution(N, stages) {
  let answer = new Array(N);
  let people = stages.length;
  answer.fill([0, 0]);
  answer = answer.map((item, idx) => [0, idx + 1]);

  stages.sort((a, b) => a - b);

  for (let i = 0; i < stages.length; i++) {
    if (stages[i] - 1 < N) {
      answer[[stages[i] - 1]][0] += 1;
    }
  }

  for (let i = 0; i < N; i++) {
    const temp = answer[i][0];
    answer[i][0] = temp / people;
    people -= temp;
  }

  answer.sort((a, b) => b[0] - a[0]);
  return answer.map((item) => item[1]);
}
